sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -1988,4 +1988,51 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       assert(errMsg.startsWith("The field for corrupt records must be string type and nullable"))
     }
   }
+
+  test("SPARK-18772: Parse special floats correctly") {
+    val jsons = Seq(
+      """{"a": "+INF"}""",
+      """{"a": "INF"}""",
+      """{"a": "-INF"}""",
+      """{"a": "NaN"}""",
+      """{"a": "Infinity"}""",
+      """{"a": "+Infinity"}""",
+      """{"a": "-Infinity"}""")
```

Comment:
Below is from `JsonParser.Feature ALLOW_NON_NUMERIC_NUMBERS`

> "INF" (for positive infinity), as well as alias of "Infinity"
> "-INF" (for negative infinity), alias "-Infinity"
> "NaN" (for other not-a-numbers, like result of division by zero)

Then, this PR also adds `+INF` and `+Infinity`. 

What is the Scala standard?